DigitalLearningSolutions.Data.Migrations/202204071530_PopulateUsersTableFromAccountsTables.cs

```diff
@@ -0,0 +1,415 @@
+                                FirstName,
+                                LastName,
+                                JobGroupID,
+                                ProfessionalRegistrationNumber,
+                                ProfileImage,
+                                Active,
+                                ResetPasswordID,
+                                TermsAgreed,
+                                FailedLoginCount,
+                                HasBeenPromptedForPrn,
+                                LearningHubAuthId,
+                                HasDismissedLhLoginWarning,
+                                EmailVerified,
+                                DetailsLastChecked
+                            )
+                            OUTPUT Inserted.ID
+                            VALUES
+                            (
+                                @primaryEmail,
+                                @passwordHash,
+                                @firstName,
+                                @lastName,
+                                @jobGroupID,
+                                @professionalRegistrationNumber,
+                                @profileImage,
+                                @active,
+                                @resetPasswordID,
+                                NULL,
+                                0,
+                                @hasBeenPromptedForPrn,
+                                @learningHubAuthId,
+                                @hasDismissedLhLoginWarning,
+                                GETUTCDATE(),
+                                GETUTCDATE()
+                            )",
+                new
+                {
+                    primaryEmail = setEmailToGuid
+                        ? Guid.NewGuid().ToString()
+                        : delegateAccount.Email,
+                    passwordHash = delegateAccount.OldPassword ?? "",
+                    firstName = delegateAccount.FirstName_deprecated ?? "",
+                    lastName = delegateAccount.LastName_deprecated,
+                    jobGroupId = delegateAccount.JobGroupId_deprecated,
+                    professionalRegistrationNumber = delegateAccount.ProfessionalRegistrationNumber_deprecated,
+                    profileImage = delegateAccount.ProfileImage_deprecated,
+                    active = delegateAccount.Active,
+                    resetPasswordId = delegateAccount.ResetPasswordId_deprecated,
+                    hasBeenPromptedForPrn = delegateAccount.HasBeenPromptedForPrn_deprecated,
+                    learningHubAuthId = delegateAccount.LearningHubAuthId_deprecated,
+                    hasDismissedLhLoginWarning = delegateAccount.HasDismissedLhLoginWarning_deprecated,
+                }
+            );
+
+            UpdateDelegateAccountUserIdEmailAndDetailsLastChecked(connection, userId, delegateAccount.Id);
+        }
+
+        private static void UpdateDelegateAccountUserIdEmailAndDetailsLastChecked(

```

Comment:
I think we can call this less, but need to finish work now. Remind me to come back to this.